### PR TITLE
Update README

### LINKED ISSUE #614: Update readme with improved logic regarding wait/e2e
The `--wait` flags were added which need to be documented better in the readme.

In addition, the `retrieve` command now outputs the names of the tarballs it creates.

Those two things together make little snippets really easy to write, something like:

```
sonobuoy run --mode quick --wait && \
sonobuoy e2e $(sonobuoy retrieve) && \
sonobuoy delete --wait
```

### PR DESCRIPTION
**Pull request by johnSchnake**
**What this PR does / why we need it**:
This changes the README in a number of ways:
 - Adds more details about quick mode which is often
used not but front and center in the README
 - Adds improved script lines which take advantate
of --wait options and capturing the filename from
retrieve
 - Adds information regarding `sonobuoy gen` which is the
best workaround whenever Sonobuoy lacks functionality users
need. Slack is usually full of questions where the answer is:
use `sonobuoy gen` and `kubectl apply`
 - Moves the link definitions to a central location at the
bottom of the file.
 - Removes prefix from code lines to improve copy/paste use
 - Clarifies installation can also be from binary which
doesnt require Go and that kubectl isn't required except in
some workflows

**Which issue(s) this PR fixes**
Fixes #614

**Special notes for your reviewer**:

**Release note**:
```
N/A
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,30 +53,49 @@ records the following results:
 $ export PATH=$GOROOT/bin:$GOPATH/bin:$PATH 
 ```  
 
-[gimme]: https://github.com/travis-ci/gimme
-
-### Download and run
+### Install
```
Comment (johnSchnake) [inline README.md]: Should probably list non-go-get installation instructions (just link to our releases/binaries). For some flows this is going to be the more common installation path.

File: README.md [hunk 2]
```diff
@@ -100,59 +107,57 @@ somewhere local:
 $ sonobuoy retrieve .
 ```
 
-This copies a single `.tar.gz` snapshot from the Sonobuoy pod into your local
+You can then extra the resulting `.tar.gz` snapshot from the Sonobuoy pod into your local
```
Comment (johnSchnake) [inline README.md]: s/extra/extract

Comment (johnSchnake) [reply README.md]: Paragraph actually needs reworked; its gotten mangled.

File: README.md [hunk 3]
```diff
@@ -100,59 +107,57 @@ somewhere local:
...
 `.` directory. Extract the contents into `./results` with:
 
 ```
+# If you only have one tar.gz in the pwd
 mkdir ./results; tar xzf *.tar.gz -C ./results
+
+# Sonobuoy retrieve echos the created file to stdout so you can capture and manipulate it.
+$ output=$(sonoobuoy retrieve)
+$ mkdir ./results; tar xzf $output -C ./results
```
Comment (johnSchnake) [inline README.md]: Side note: I wish github allowed multi-language snippets. I always feel bad listing directions for Mac/Linux without having a Windows version.

File: README.md [hunk 4]
```diff
@@ -20,104 +38,162 @@ for the following use cases:
...
 
-Deploy a Sonobuoy pod to your cluster with:
+This runs only the e2e plugin (Kubernetes tests) but specifies that it should
+only run a single, short test. This should normally take less than a minute to complete.
 
+### Monitoring Sonobuoy during a run
+You can check on the status of each of the plugins running with:
 ```
-$ sonobuoy run
+sonobuoy status
```
Comment (johnSchnake) [inline README.md]: See https://github.com/civicrm/civicrm-dev-docs/issues/190 for a random discussion of `$` prefix vs not. I think they hit on the relevant points and I liked where they landed.

tl;dr; No prefix unless mixing input/output where you need to distinguish them.

**Comment by johnSchnake** [comment]
@stevesloka I originally started with the desire to add details about `--wait` and such but I started finding more things that could use some TLC. Let me know what you think.

Being a readme, it would be great to get some broader community feedback on this too.

File: README.md (hunk 4 above)
Comment (stevesloka) [reply README.md]: I always put in the prefix and on your issue, they revised to always use: https://github.com/civicrm/civicrm-dev-docs/issues/190#issuecomment-322019933 

I don't feel super strongly either way, but like how putting it in calls out that it's something you run in a terminal.

File: README.md [hunk 5]
```diff
@@ -1,7 +1,25 @@
 # <img src="docs/img/sonobuoy-logo.png" width="400px" > [![Build Status][status]][travis]
 
-[status]: https://travis-ci.org/heptio/sonobuoy.svg?branch=master
-[travis]: https://travis-ci.org/heptio/sonobuoy/#
+- [Overview](#overviewoview)
```
Comment (stevesloka) [inline README.md]: I think the TOC might be too much for a front-page readme. We should get to the point and explain what Sonobuoy is, how it works, and maybe a quick, getting started. I think anything more than that we should move off to other docs pages and dig in deeper.

File: README.md [hunk 6]
```diff
@@ -20,104 +38,162 @@ for the following use cases:
...
-* systemd logs from each host. Requires a plugin.
-* The results of a e2e conformance tests.
+Golang version 1.12 or greater is recommended. Golang can be installed via
+[gimme][gimme].
 
-### CLI Prerequisites
+## Getting Started
+To launch Sonobuoy for a short run and wait until it is finished run:
+```
+sonobuoy run --mode quick --wait
```
Comment (stevesloka) [inline README.md]: I think this should remove the `-m quick` and just do the regular scan. That can then merge this section with "checking for Cluster Conformance".

File: README.md [hunk 7]
```diff
@@ -20,104 +38,162 @@ for the following use cases:
...
+To run Sonobuoy to check if your cluster passes Kubernetes conformance, run:
+```
+sonobuoy run
+```
+
+This is the command which will generate the data necessary to validate your
+cluster is conformant according to the [CNCF][cncf].
 
-### Download and run
+### Quick Mode
```
Comment (stevesloka) [inline README.md]: I would ditch this whole section

File: README.md [hunk 8]
```diff
@@ -20,104 +38,162 @@ for the following use cases:
...
+option.
 
 ```
-mkdir ./results; tar xzf *.tar.gz -C ./results
+sonobuoy delete --all --wait=5
 ```
 
-For information on the contents of the snapshot, see the [snapshot
-documentation][snapshot].
+## Advanced use cases
```
Comment (stevesloka) [inline README.md]: I would move this whole section into `docs`

File: README.md [hunk 9]
```diff
@@ -20,104 +38,162 @@ for the following use cases:
...
+
+Delete the main namespace and cluster resources by running:
 ```
-$ sonobuoy retrieve .
+sonobuoy delete 
 ```
 
-This copies a single `.tar.gz` snapshot from the Sonobuoy pod into your local
-`.` directory. Extract the contents into `./results` with:
+There are some Kubernetes e2e tests that may leak some resources. Sonobuoy can
```
Comment (stevesloka) [inline README.md]: Move this to `docs` section. Also, needs to make sure to call out how it does delete with `--all` (i.e. it deletes any namespace starting with `e2e*`)

File: README.md [hunk 10]
```diff
@@ -20,104 +38,162 @@ for the following use cases:
...
 ```
 
-To view the output, copy the output directory from the main Sonobuoy pod to
-somewhere local:
+For information on the contents of the snapshot, see the [snapshot
+documentation][snapshot].
+
+### Cleanup
+
+Sonobuoy creates a few resources in order to run and expects to run within its
```
Comment (stevesloka) [inline README.md]: Merge this text with the "getting started section"

File: README.md (hunk 4 above)
Comment (johnSchnake) [reply README.md]: Haha, apparently I stopped reading at the first summary when it matched my feelings 😅 I can do either; I wish github markdown had a way to put a non selectable image there instead of text so you could see it but it wouldn’t bother copy paste.

Comment (stevesloka) [reply README.md]: haha, we just need to be consistent. Looks like velero doesn't use them, but I like how it looks when you show a command with output, to distinguish the command from the output. ¯\_(ツ)_/¯

Comment (johnSchnake) [reply README.md]: 👍 I'm at least being consistent here; not using them throughout. I only do show single line commands and am not mixing with output so it doesnt really matter in the end now that I've fixed up some other things.

File: README.md (hunk 5 above)
Comment (johnSchnake) [reply README.md]: TOC removed and readme made shorter

File: README.md (hunk 6 above)
Comment (johnSchnake) [reply README.md]: Did this and removed the quick scan section. Just added a note below this mentioning it as a much shorter alternative for general configuration check.

File: README.md (hunk 8 above)
Comment (johnSchnake) [reply README.md]: Did this more or less. Added new docs pages but also linked to them and changed the `more info` section below. I think it looks/flows well and is more likely to get clicks since the topics are more visible and require fewer clicks to get to.

File: README.md (hunk 9 above)
Comment (johnSchnake) [reply README.md]: Moved to a known issue section and shortened.

**Comment by johnSchnake** [comment]
@stevesloka I think I addressed all of your concerns, let me know what you think. Since the last review:
 - removed TOC
 - shortened readme
 - moved some of readme into docs pages
 - changed/reworded the more info section
 - a few other minor changes
